hi-ml/src/health_ml/model_trainer.py

```diff
@@ -255,6 +260,17 @@ def model_train(checkpoint_path: Optional[Path],
     assert trainer.logger is not None
     trainer.logger.finalize('success')
 
+    if container.additional_val_epoch and hasattr(lightning_model, "additional_val_epoch"):
```

Comment:
I'm not sure the `hasattr` check is a good idea. It is also nowhere documented that you need to have the flag to run the extra val epoch. I'd stick with just the flag at container level.

Reply:
This is to safely execute line 271. I can maybe make it an assert and document that it's needed to activate extra outputs at validation time, what do you think